app/services/reachable_users_service.rb

```diff
@@ -10,17 +10,22 @@ def get_users(limit = nil)
     get_users_with_random(limit)
   end
 
+  def get_vaccination_center_grid_query
+    cells = ::GridCoordsService.new(@vaccination_center.lat, @vaccination_center.lon).get_covering(@campaign.max_distance_in_meters)[:cells]
+    "(grid_i, grid_j) IN ((" + cells.map { |sub| sub.join(",") }.join("),(") + "))"
+  end
+
   def get_users_with_v2(limit = nil)
     sql = <<~SQL.tr("\n", " ").squish
       with reachable_users as (
         SELECT
         u.id as user_id,
         (SQRT((((:lat) - u.lat)*110.574)^2 + (((:lon) - u.lon)*111.320*COS(u.lat::float*3.14159/180))^2)) as distance
```

Comment:
Should we also use the grid here to compute a distance we can use to rank ?

Reply:
Thought we could do that in another PR, but can do that here if you prefer

Reply:
It works too in another PR

Reply:
I just feel this where we will gain the most performance

Reply:
I'll add it.

I believe the most speedup will be in the `WHERE` clause as it is no longer scanning on all possible candidates but using the index to run the query.

Reply:
it's done